pkg/common/client.go

```diff
@@ -98,6 +98,49 @@ func (c *Client) CreateClientRole(clientID string, role *v1alpha1.RoleRepresenta
 	return c.create(role, fmt.Sprintf("realms/%s/clients/%s/roles", realmName, clientID), "client role")
 }
 
+func (c *Client) ListRealmRoles(realmName string) ([]*v1alpha1.RoleRepresentation, error) {
+	result, err := c.list(fmt.Sprintf("realms/%s/roles", realmName), "Realm Roles", func(body []byte) (T, error) {
+		var realmRoles []*v1alpha1.RoleRepresentation
+
+		err := json.Unmarshal(body, &realmRoles)
+		return realmRoles, err
+	})
+
```

Comment:
```suggestion
	if err != nil {
		return nil, err
	}
```
Line `109` is casting the generic pointer into a specific type (slice of RoleRepresentation) and if there is an http error, line `102` will return `Nil` and an `err`. If you don't check the `err` first, you will thrown an exception trying to cast Nil into RoleRepresentation. This is done in some upstream functions, but not others. I think we should do it in all your new "List" calls, as it is a safer approach.
Examples: `ListRealmRoleClientRoleComposites` and `ListClients`